program-runtime/src/loaded_programs.rs

```diff
@@ -76,14 +87,8 @@ pub enum LoadedProgramType {
     ///
     /// It continues to track usage statistics even when the compiled executable of the program is evicted from memory.
     Unloaded(ProgramRuntimeEnvironment),
-    /// Verified and compiled program of loader-v1 or loader-v2
-    LegacyV0(Executable<InvokeContext<'static>>),
-    /// Verified and compiled program of loader-v3 (aka upgradable loader)
-    LegacyV1(Executable<InvokeContext<'static>>),
-    /// Verified and compiled program of loader-v4
-    Typed(Executable<InvokeContext<'static>>),
-    #[cfg(test)]
-    TestLoaded(ProgramRuntimeEnvironment),
+    /// Verified and compiled program
+    Loaded(Executable<InvokeContext<'static>>),
```

Comment:
maybe not to address in this PR, but I think the enum name is misnomer.  A LoadedProgramType value can be 'Loaded', 'Unloaded', or something else. It doesn't make sense when a 'loaded program' is in fact 'unloaded'. It also doesn't make sense that for a LoadedProgramType you add an enum value 'Loaded'.  If only 'Loaded' programs are loaded, then what are other 'LoadedProgramType' values?

Reply:
Definitely, sounds better. I don't insist that these name changes should be in this PR. It's up to you.